src/theory/quantifiers/ceg_instantiator.cpp

```diff
@@ -46,25 +46,42 @@ CegInstantiator::~CegInstantiator() {
 void CegInstantiator::computeProgVars( Node n ){
   if( d_prog_var.find( n )==d_prog_var.end() ){
     d_prog_var[n].clear();
-    if( std::find( d_vars.begin(), d_vars.end(), n )!=d_vars.end() ){
-      d_prog_var[n][n] = true;
+    if (n.getKind() == kind::CHOICE)
+    {
+      Assert(d_prog_var.find(n[0][0]) == d_prog_var.end());
+      d_prog_var[n[0][0]].clear();
+    }
+    if (d_vars_set.find(n) != d_vars_set.end())
+    {
+      d_prog_var[n].insert(n);
     }else if( !d_out->isEligibleForInstantiation( n ) ){
-      d_inelig[n] = true;
+      d_inelig.insert(n);
       return;
     }
     for( unsigned i=0; i<n.getNumChildren(); i++ ){
       computeProgVars( n[i] );
       if( d_inelig.find( n[i] )!=d_inelig.end() ){
-        d_inelig[n] = true;
+        d_inelig.insert(n);
       }
-      for( std::map< Node, bool >::iterator it = d_prog_var[n[i]].begin(); it != d_prog_var[n[i]].end(); ++it ){
-        d_prog_var[n][it->first] = true;
-      }
-      //selectors applied to program variables are also variables
-      if( n.getKind()==APPLY_SELECTOR_TOTAL && d_prog_var[n].find( n[0] )!=d_prog_var[n].end() ){
-        d_prog_var[n][n] = true;
+      // all variables in child are cotained in this
+      for (std::unordered_set<Node, NodeHashFunction>::iterator it =
+               d_prog_var[n[i]].begin();
+           it != d_prog_var[n[i]].end();
+           ++it)
+      {
+        d_prog_var[n].insert(*it);
```

Comment:
Can you use
```
d_prog_var[n] = std::unordered_set<Node, NodeHashFunction>(d_prog_var[n[i]].begin(), d_prog_var[n[i]].end())
```
here?

Reply:
This needs to append, not overwrite.  Does d_prog_var[n].insert(d_prog_var[n[i]].begin(), d_prog_var[n[i]].end()) work in this case?

Reply:
Yes this should work.